framework/math/dense_matrix.h

```diff
@@ -415,7 +415,7 @@ GaussElimination(DenseMatrix<TYPE>& A, Vector<TYPE>& b, unsigned int n)
   }
 
   // Back substitution
-  for (int i = n - 1; i >= 0; --i)
+  for (auto i = n - 1; i >= 0; --i)
```

Comment:
For this for-loop to work, `i` must become negative (that's the stopping condition). However, the type is inferred from `n` which is `unsigned int`, so `i` will never be negative.

You have 3 options:
1. change type of `n`
2. keep the explicit `int i`
3. rewrite the loop so that `i` is only positive.

This is most likely what is causing the test suite to fail.

Reply:
I would suggest option (2). Checking against 0 is going to be more efficient than (3), and this code is particularly performance sensitive.